### PR TITLE
fix(bug): Fix KB configuration

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by AssanDiomande — DISMISSED** [review]

File: www/include/Administration/parameters/DB-Func.php [hunk 1]
```diff
@@ -1040,12 +1040,25 @@ function updateBackupConfigData($db, $form, $centreon)
...
+            $ret["kb_wiki_password"] = $db->escape($ret["kb_wiki_password"]);
+        }
+    }
+
+    if (isset($ret["kb_wiki_url"]) && !filter_var($ret["kb_wiki_url"], FILTER_VALIDATE_URL)) {
+        unset($ret["kb_wiki_url"]);
+    }
+
+    if (isset($ret["kb_wiki_account"])) {
+        $ret["kb_wiki_account"] = $db->escape($ret["kb_wiki_account"]);
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:1061]: juste prepare query in `updateOption`, it will avoid to use escape

Comment (kduret) [reply www/include/Administration/parameters/DB-Func.php:1061]: cause the injection issue seems to be here :)
```php
function updateOption($pearDB, $key, $value)
{
    /*
     * Purge
     */
    $pearDB->query("DELETE FROM `options` WHERE `key` = '$key'");

    /*
     * Add
     */
    if (!is_null($value) && $value != 'NULL') {
        $value = "'$value'";
    }
    $pearDB->query("INSERT INTO `options` (`key`, `value`) VALUES ('$key', $value)");
}
```

File: www/include/Administration/parameters/DB-Func.php [hunk 2]
```diff
@@ -34,20 +34,23 @@
  *
  */
 
+ /**
+ * Used to update fields in the 'centreon.options' table
+ *
+ * @param object $pearDB : database connection
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:40]:
```suggestion
 * @param \CentreonDB $pearDB : database connection
```

File: www/include/Administration/parameters/DB-Func.php [hunk 3]
```diff
@@ -1040,12 +1043,25 @@ function updateBackupConfigData($db, $form, $centreon)
...
+        }
+    }
+
+    if (isset($ret["kb_wiki_url"]) && !filter_var($ret["kb_wiki_url"], FILTER_VALIDATE_URL)) {
+        unset($ret["kb_wiki_url"]);
+    }
+
+    if (isset($ret["kb_wiki_account"])) {
+        $ret["kb_wiki_account"] = $db->escape($ret["kb_wiki_account"]);
     }
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:1065]:
```suggestion
```

File: www/include/Administration/parameters/DB-Func.php [hunk 4]
```diff
@@ -1040,12 +1043,25 @@ function updateBackupConfigData($db, $form, $centreon)
...
 
-    if (isset($ret["kb_wiki_password"]) && $ret["kb_wiki_password"] === CentreonAuth::PWS_OCCULTATION) {
-        unset($ret["kb_wiki_password"]);
+    if (isset($ret["kb_wiki_password"])) {
+        if ($ret["kb_wiki_password"] === CentreonAuth::PWS_OCCULTATION) {
+            unset($ret["kb_wiki_password"]);
+        } else {
+            $ret["kb_wiki_password"] = $db->escape($ret["kb_wiki_password"]);
+        }
+    }
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:1057]: retrieve previous code

File: www/include/Administration/parameters/DB-Func.php (hunk 2 above)
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:40]:
```suggestion
 * @param \CentreonDB $pearDB : database connection
```

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**